src/utils/auth_token/auth_token.rs

```diff
@@ -0,0 +1,98 @@
+//! Defines the AuthToken type, which stores a Sentry auth token.
+
+use super::AuthTokenPayload;
+use super::{OrgAuthToken, UserAuthToken};
+use std::fmt::{Display, Formatter, Result};
+
+/// Represents a (soft) validated Sentry auth token.
+#[derive(Debug, Clone)]
+pub struct AuthToken(AuthTokenInner);
+
+impl AuthToken {
+    /// Constructs a new AuthToken from a string. Logs a warning if the auth token's
+    /// format is unrecognized.
+    fn new(auth_string: String) -> Self {
+        AuthToken(AuthTokenInner::new(auth_string))
+    }
+
+    /// Returns the payload of the auth token, if it is an org auth token.
+    pub fn payload(&self) -> Option<&AuthTokenPayload> {
+        self.0.payload()
+    }
+}
+
+impl From<String> for AuthToken {
+    /// Constructs a new AuthToken from a string. Logs a warning if the auth token's
+    /// format is unrecognized.
+    fn from(auth_string: String) -> Self {
+        AuthToken::new(auth_string)
+    }
+}
+
+impl<'a> From<&'a AuthToken> for &'a str {
+    /// Retrieves a reference to the auth token string from an AuthToken.
+    fn from(auth_token: &'a AuthToken) -> Self {
+        let ref inner = auth_token.0;
+        inner.into()
+    }
+}
```

Comment:
This `From` impl is very weird, and you usually do these using a `as_str` fn.
On the other way around, you should probably have a `From<&str> for AuthToken` as well.

Reply:
I think I'll just go with an `as_str` function